pkg/chunk/schema_config.go

```diff
@@ -70,111 +69,11 @@ type SchemaConfig struct {
-	f.Var(&cfg.V4SchemaFrom, "dynamodb.v4-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v4 schema.")
-	f.Var(&cfg.V5SchemaFrom, "dynamodb.v5-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v5 schema.")
-	f.Var(&cfg.V6SchemaFrom, "dynamodb.v6-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v6 schema.")
-	f.Var(&cfg.V9SchemaFrom, "dynamodb.v9-schema-from", "The date (in the format YYYY-MM-DD) after which we enable v9 schema (Series indexing).")
-	f.Var(&cfg.BigtableColumnKeyFrom, "bigtable.column-key-from", "The date (in the format YYYY-MM-DD) after which we use bigtable column keys.")
-
-	f.StringVar(&cfg.OriginalTableName, "dynamodb.original-table-name", "cortex", "The name of the DynamoDB table used before versioned schemas were introduced.")
-	f.BoolVar(&cfg.UsePeriodicTables, "dynamodb.use-periodic-tables", false, "Should we use periodic tables.")
-
-	f.Var(&cfg.IndexTablesFrom, "dynamodb.periodic-table.from", "Date after which to use periodic tables.")
-	cfg.IndexTables.RegisterFlags("dynamodb.periodic-table", "cortex_", f)
-	f.Var(&cfg.ChunkTablesFrom, "dynamodb.chunk-table.from", "Date after which to write chunks to DynamoDB.")
-	cfg.ChunkTables.RegisterFlags("dynamodb.chunk-table", "cortex_chunks_", f)
-}
-
-func (cfg *SchemaConfig) loadFromFlags() error {
-	cfg.Configs = []PeriodConfig{}
-
-	add := func(t string, f model.Time) {
-		cfg.Configs = append(cfg.Configs, PeriodConfig{
-			From:      DayTime{f},
-			Schema:    t,
-			IndexType: cfg.legacy.StorageClient,
-			IndexTables: PeriodicTableConfig{
-				Prefix: cfg.legacy.OriginalTableName,
-				Tags:   cfg.legacy.IndexTables.Tags,
-			},
-		})
-	}
-
-	add("v1", 0)
-
-	if cfg.legacy.DailyBucketsFrom.IsSet() {
-		add("v2", cfg.legacy.DailyBucketsFrom.Time)
-	}
-	if cfg.legacy.Base64ValuesFrom.IsSet() {
-		add("v3", cfg.legacy.Base64ValuesFrom.Time)
-	}
-	if cfg.legacy.V4SchemaFrom.IsSet() {
-		add("v4", cfg.legacy.V4SchemaFrom.Time)
-	}
-	if cfg.legacy.V5SchemaFrom.IsSet() {
-		add("v5", cfg.legacy.V5SchemaFrom.Time)
-	}
-	if cfg.legacy.V6SchemaFrom.IsSet() {
-		add("v6", cfg.legacy.V6SchemaFrom.Time)
-	}
-	if cfg.legacy.V9SchemaFrom.IsSet() {
-		add("v9", cfg.legacy.V9SchemaFrom.Time)
-	}
-
-	cfg.ForEachAfter(cfg.legacy.IndexTablesFrom.Time, func(config *PeriodConfig) {
-		config.IndexTables = cfg.legacy.IndexTables
-	})
-	if cfg.legacy.ChunkTablesFrom.IsSet() {
-		cfg.ForEachAfter(cfg.legacy.ChunkTablesFrom.Time, func(config *PeriodConfig) {
-			if config.IndexType == "aws" {
-				config.IndexType = "aws-dynamo"
-			}
-			config.ChunkTables = cfg.legacy.ChunkTables
-		})
-	}
-	if cfg.legacy.BigtableColumnKeyFrom.IsSet() {
-		cfg.ForEachAfter(cfg.legacy.BigtableColumnKeyFrom.Time, func(config *PeriodConfig) {
-			config.IndexType = "gcp-columnkey"
-		})
-	}
-	return nil
+	flag.StringVar(&cfg.fileName, "schema-config-file", "", "The path to the schema config file.")
```

Comment:
Suggest supporting the old name as a deprecated option.

Reply:
Done, also tested that it works.